accepted/file-enumeration.md

```diff
@@ -0,0 +1,238 @@
+# Extensible File Enumeration
+
+**PM** [Immo Landwerth](https://github.com/terrajobst) | **Dev** [Jeremy Kuhne](https://github.com/jeremykuhne)
+
+Enumerating files in .NET provides limited configurability. You can specify a simple DOS style pattern and whether or not to look recursively. More complicated filtering requires post filtering all results which can introduce a significant performance drain.
+
+Recursive enumeration is also problematic in that there is no way to handle error states such as access issues or cycles created by links.
+
+These restrictions have a significant impact on file system intensive applications, a key example being MSBuild. We can address these restrictions and provide performant, configurable enumeration.
+
+## Scenarios and User Experience
+
+1. MSBuild can custom filter filesystem entries with limited allocations and form the results in any desired format.
+2. Users can build custom enumerations utilizing completely custom or provided commonly used filters and transforms.
+
+## Requirements
+
+1. Filtering based on common file system data is possible
+	a. Name
+	b. Attributes
+	c. Time stamps
+    d. File size
+2. Result transforms can be of any type
+3. We provide common filters and transforms
+4. Recursive behavior is configurable
+5. Error handling is configurable
+
+### Goals
+
+1. API is cross platform generic
+2. API minimizes allocations while meeting #1
+
+### Non-Goals
+
+1. API will not expose platform specific data
+3. Error handling configuration is fully customizable
```

Comment:
>Error handling configuration is fully customizable [](start = 3, length = 50)

That needs more detail as a goal says *Error handling is configurable* while a non-goal says *Error handling configuration is fully customizable*. You need to draw that line.

Reply:
Got it

---
In reply to: [154807326](https://github.com/dotnet/designs/pull/24#discussion_r154807326) [](ancestors = 154807326)